data-reporting-service/common-util/src/main/java/gov/cdc/etldatapipeline/commonutil/json/CustomJsonGeneratorImpl.java

```diff
@@ -0,0 +1,110 @@
+package gov.cdc.etldatapipeline.commonutil.json;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.PropertyNamingStrategies;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;
+import com.google.common.base.CaseFormat;
+import gov.cdc.etldatapipeline.commonutil.model.DataRequiredFields;
+import gov.cdc.etldatapipeline.commonutil.model.avro.DataEnvelope;
+import gov.cdc.etldatapipeline.commonutil.model.avro.DataField;
+import gov.cdc.etldatapipeline.commonutil.model.avro.DataSchema;
+
+import java.lang.reflect.Field;
+import java.util.HashSet;
+import java.util.Set;
+
+public class CustomJsonGeneratorImpl {
+    public <T extends DataRequiredFields> DataEnvelope buildAvroRecord(T obj) {
```

Comment:
We might need to do integration testing with the jdbc sink connector. The avro record generation with the jdbc sink connector.

Reply:
I'll create a separate story for integration tests